addons/stock/models/stock_quant.py

```diff
@@ -96,39 +95,34 @@ def _domain_product_id(self):
     tracking = fields.Selection(related='product_id.tracking', readonly=True)
     on_hand = fields.Boolean('On Hand', store=False, search='_search_on_hand')
 
+    # Inventory Fields
+    inventory_quantity = fields.Float(
+        'Inventoried Quantity', digits='Product Unit of Measure', groups='stock.group_stock_manager',
+        help="The product's counted quantity.")
+    inventory_diff_quantity = fields.Float(
+        'Difference', compute='_compute_inventory_diff_quantity', store=True,
+        help="Indicates the gap between the product's theoretical quantity and its counted quantity.",
+        readonly=True, digits='Product Unit of Measure', groups='stock.group_stock_manager')
+    inventory_date = fields.Date(
+        'Inventory Date', compute='_compute_inventory_date', store=True, readonly=False,
+        help="Next date the On Hand Quantity should be counted.")
+    user_id = fields.Many2one(
+        'res.users', 'User', groups='stock.group_stock_manager', help="User assigned to do product count.")
+
     @api.depends('quantity', 'reserved_quantity')
     def _compute_available_quantity(self):
         for quant in self:
             quant.available_quantity = quant.quantity - quant.reserved_quantity
 
-    @api.depends('quantity')
-    def _compute_inventory_quantity(self):
-        if not self._is_inventory_mode():
-            self.inventory_quantity = 0
-            return
-        for quant in self:
-            quant.inventory_quantity = quant.quantity
+    @api.depends('location_id')
+    def _compute_inventory_date(self):
+        for quants in self.filtered(lambda q: not q.inventory_date):
+            quants._set_inventory_date()
```

Comment:
I would leave the code entirely in the _compute (unless you use it elsewhere but I did not found it)

Reply:
Dang, forgot to finish putting it in `_apply_inventory`. It's the same logic, hence why I tried to split it out. it's also why there's the write in `_set_inventory_date` (there's another write that comes after it)